// plugin-welcome/src/api/models/EntDoctorinfo.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDoctorinfoEdges,
    EntDoctorinfoEdgesFromJSON,
    EntDoctorinfoEdgesFromJSONTyped,
    EntDoctorinfoEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntDoctorinfo
 */
export interface EntDoctorinfo {
    /**
     * Doctorname holds the value of the "doctorname" field.
     * @type {string}
     * @memberof EntDoctorinfo
     */
    doctorname?: string;
    /**
     * Doctorsurname holds the value of the "doctorsurname" field.
     * @type {string}
     * @memberof EntDoctorinfo
     */
    doctorsurname?: string;
    /**
     * 
     * @type {EntDoctorinfoEdges}
     * @memberof EntDoctorinfo
     */
    edges?: EntDoctorinfoEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntDoctorinfo
     */
    id?: number;
    /**
     * Licensenumber holds the value of the "licensenumber" field.
     * @type {string}
     * @memberof EntDoctorinfo
     */
    licensenumber?: string;
    /**
     * Telephonenumber holds the value of the "telephonenumber" field.
     * @type {string}
     * @memberof EntDoctorinfo
     */
    telephonenumber?: string;
}

export function EntDoctorinfoFromJSON(json: any): EntDoctorinfo {
    return EntDoctorinfoFromJSONTyped(json, false);
}

export function EntDoctorinfoFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntDoctorinfo {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'doctorname': !exists(json, 'doctorname') ? undefined : json['doctorname'],
        'doctorsurname': !exists(json, 'doctorsurname') ? undefined : json['doctorsurname'],
        'edges': !exists(json, 'edges') ? undefined : EntDoctorinfoEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'licensenumber': !exists(json, 'licensenumber') ? undefined : json['licensenumber'],
        'telephonenumber': !exists(json, 'telephonenumber') ? undefined : json['telephonenumber'],
    };
}

export function EntDoctorinfoToJSON(value?: EntDoctorinfo | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'doctorname': value.doctorname,
        'doctorsurname': value.doctorsurname,
        'edges': EntDoctorinfoEdgesToJSON(value.edges),
        'id': value.id,
        'licensenumber': value.licensenumber,
        'telephonenumber': value.telephonenumber,
    };
}
