// plugin-welcome/src/api/models/EntPaytypeEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPaytypeEdges
 */
export interface EntPaytypeEdges {
    /**
     * Bills holds the value of the bills edge.
     * @type {Array<EntBill>}
     * @memberof EntPaytypeEdges
     */
    bills?: Array<EntBill>;
}

export function EntPaytypeEdgesFromJSON(json: any): EntPaytypeEdges {
    return EntPaytypeEdgesFromJSONTyped(json, false);
}

export function EntPaytypeEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPaytypeEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'bills': !exists(json, 'bills') ? undefined : ((json['bills'] as Array<any>).map(EntBillFromJSON)),
    };
}

export function EntPaytypeEdgesToJSON(value?: EntPaytypeEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'bills': value.bills === undefined ? undefined : ((value.bills as Array<any>).map(EntBillToJSON)),
    };
}
